src/components/Popup/Popup.tsx

```diff
@@ -96,56 +109,97 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
     position: 'above',
+    accessibility: PopupBehavior as Accessibility,
   }
 
-  public state = { triggerRef: undefined }
+  public state = { triggerRef: undefined, popupOpened: false }
+
+  actionHandlers: AccessibilityActionHandlers = {
+    performClick: e => this.onTriggerClick(e),
+    closeAndFocusTrigger: e => this.handlePopupState(e, false, this.focusTrigger),
+  }
 
   public renderComponent({
     ElementType,
     classes,
+    accessibility,
     rest,
     rtl,
   }: IRenderResultConfig<IPopupProps>): React.ReactNode {
-    const { children, trigger } = this.props
+    const { children, trigger, position, align } = this.props
 
     return (
       <ElementType
         className={classes.root}
         {...rest}
+        open={this.state.popupOpened}
         trigger={trigger}
         triggerRef={this.handleTriggerRef}
+        triggerAccessibility={{
+          ...accessibility.attributes.trigger,
+          ...accessibility.keyHandlers.trigger,
+        }}
+        onOutsideClick={e => this.handlePopupState(e, false)}
+        onTriggerClick={this.onTriggerClick}
       >
-        {childrenExist(children) ? children : this.renderContent(rtl)}
+        {childrenExist(children)
+          ? children
+          : this.renderContent(
+              computePopupPlacement({ align, position, rtl }),
+              this.renderPopperChildren.bind(this, rtl, accessibility),
+            )}
       </ElementType>
     )
   }
 
-  private renderContent(rtl: boolean): JSX.Element {
-    const { align, position } = this.props
+  private renderContent(
+    popupPlacement: Placement,
+    renderChildrenCb: (props: PopperChildrenProps) => React.ReactNode,
+  ): JSX.Element {
     const triggerRef = this.state.triggerRef
-    const placement = computePopupPlacement({ align, position, rtl })
 
     return (
       triggerRef && (
         <Popper
-          placement={placement}
+          placement={popupPlacement}
           referenceElement={triggerRef}
-          children={this.renderPopperChildren.bind(this, rtl)}
+          children={renderChildrenCb}
         />
       )
     )
   }
 
-  private renderPopperChildren = (rtl: boolean, { ref, style }: PopperChildrenProps) => {
+  private renderPopperChildren = (
+    rtl: boolean,
+    accessibility: IAccessibilityBehavior,
+    { ref, style }: PopperChildrenProps,
+  ) => {
     const { basic, content } = this.props
     const computedStyle = rtl ? rtlCSSJS(style) : style
 
     return (
-      <Popup.Content innerRef={ref} basic={basic} {...rtl && { dir: 'rtl' }} styles={computedStyle}>
+      <Popup.Content
+        innerRef={ref}
+        basic={basic}
+        {...rtl && { dir: 'rtl' }}
+        styles={computedStyle}
+        {...accessibility.attributes.popup}
+        {...accessibility.keyHandlers.popup}
+      >
         {content}
       </Popup.Content>
     )
   }
 
+  private onTriggerClick = (e: Event) => {
+    this.handlePopupState(e, !this.state.popupOpened)
```

Comment:
`state` could be updated asynchronously in React - for that reason I would suggest to use the following form of `setState` method in `handlePopupState`:

```jsx
 private handlePopupState = (..., getPopupOpened: (previousOpened) => boolean, ...) => {
    ..
    this.setState((previousState) => { popupOpened: getPopupOpened(previousState.popupOpened) }, ...)
  }
```